### ISSUE TITLE
New configure script does not take into account compile flags

### ISSUE THREAD

**Issue by mremond** [processone/xml#14]
With the new configure script macro, it is only possible to pass specific expat version with configure option:

    ./configure --with-expat=/custom/lib

If you export LDFLAGS, CPPFLAGS and CFLAGS but do not use `--with-expat` option, they are ignored and expat is only search searched in standard places: `/usr` and `/usr/local`

However, on OSX El Capitan with [Homebrew](http://brew.sh) expat is installed in `/usr/local/opt/expat`. It means the following configure for p1_xml works:

    ./configure --with-expat=/usr/local/opt/expat
    ...
    checking for Expat XML Parser headers in /usr/local/opt/expat/include... found
    checking for Expat XML Parser libraries... found
    checking for Expat XML Parser... yes
    checking if Expat XML Parser version is >= 2.0.0... yes
    configure: creating ./config.status
    config.status: creating vars.config

But not:

    LDFLAGS="-L/usr/local/opt/expat/lib" CFLAGS="-I/usr/local/opt/expat/include" CPPFLAGS="-I/usr/local/opt/expat/include" ./configure
    ...
    checking for Expat XML Parser... no
    configure: error: Expat XML Parser was not found

However, when building `xml` library from ejabberd build chain, we can only rely on compile flags define in the environment. It means ejabberd cannot be built  with OS X El Capitan and Homebrew as is due to this issue.

Relates to: #11

**Comment by leoliu** [comment]
compile `ejabberd` from master also fails on centos7 with stock `expat` and `expat-devel`.
...
checking whether c++ accepts -g... no
checking for Expat XML Parser headers in /usr/include... not found
checking for Expat XML Parser... no
configure: error: Expat XML Parser was not found
ERROR: Command ['get-deps'] failed!
make: *** [deps/.got] Error 1
...

**Comment by earthspike** [comment]
Same on Ubuntu 14.04. libexpat1-dev installed, /usr/include/expat.h exists and is world-readable, yet the same error as leoliu above.

```
git clone https://github.com/processone/ejabberd.git
cd ejabberd
./autogen.sh 
./configure --enable-user=ejabberd
make
```
also yields
```
checking whether c++ accepts -g... no
checking for Expat XML Parser headers in /usr/include... not found
checking for Expat XML Parser... no
configure: error: Expat XML Parser was not found
ERROR: Command ['get-deps'] failed!
make: *** [deps/.got] Error 1
```
Then I did this:
```
cd deps/p1_xml/
./configure --with-expat-inc=/usr/include/ --with-expat-lib=/usr/lib
```
and again got
```
checking whether g++ accepts -g... no
checking for Expat XML Parser headers in /usr/include... not found
checking for Expat XML Parser... no
configure: error: Expat XML Parser was not found
```
So it doesn't look like the passing of parameters/env vars is the issue, but the detection in p1_xml??

**Comment by mremond** [comment]
Fixed by 50e260704bc6089c48a4e8afbf762df87909c5df
